lib/ansible/executor/task_executor.py

```diff
@@ -224,7 +225,15 @@ def _get_loop_items(self):
                 loop_terms = listify_lookup_plugin_terms(terms=self._task.loop, templar=templar, loader=self._loader, fail_on_undefined=fail,
                                                          convert_bare=False)
                 if not fail:
-                    loop_terms = [t for t in loop_terms if not templar.is_template(t)]
+                    intermediate = []
+                    for term in loop_terms:
+                        if isinstance(term, Mapping) and 'files' in term:
+                            term['files'] = [t for t in term['files'] if not templar.is_template(t)]
+                            intermediate.append(term)
+                        else:
+                            if not templar.is_template(term):
```

Comment:
These 2 lines can probably just be collapsed into an `elif`

Reply:
done, I've just c/p without looking hardly on it.